src/coreclr/jit/lsraarm64.cpp

```diff
@@ -1011,40 +1011,49 @@ int LinearScan::BuildHWIntrinsic(GenTreeHWIntrinsic* intrinsicTree)
 
     if (intrin.op1 != nullptr)
     {
-        bool simdRegToSimdRegMove = false;
-
-        if ((intrin.id == NI_Vector64_CreateScalarUnsafe) || (intrin.id == NI_Vector128_CreateScalarUnsafe))
-        {
-            simdRegToSimdRegMove = varTypeIsFloating(intrin.op1);
-        }
-        else if (intrin.id == NI_AdvSimd_Arm64_DuplicateToVector64)
+        // Do not give a register to the second operand if it is contained.
+        if (!hasImmediateOperand && intrin.op1->isContained())
         {
-            simdRegToSimdRegMove = (intrin.op1->TypeGet() == TYP_DOUBLE);
+            assert(HWIntrinsicInfo::SupportsContainment(intrin.id));
+            assert(intrin.op1->IsVectorZero());
         }
-        else if ((intrin.id == NI_Vector64_ToScalar) || (intrin.id == NI_Vector128_ToScalar))
+        else
         {
-            simdRegToSimdRegMove = varTypeIsFloating(intrinsicTree);
-        }
+            bool simdRegToSimdRegMove = false;
```

Comment:
nit: The diff here could be a lot smaller if `simdRegToSimdRegMove` was kept above the first if block. It will be unused in one of the paths, but that should be fine.

Reply:
I'll see if I can make it a bit better.